http-server/src/main/java/io/airlift/http/server/HttpRequestEvent.java

```diff
@@ -42,8 +43,8 @@ public static HttpRequestEvent createHttpRequestEvent(Request request, Response
             user = principal.getName();
         }
 
-        String token = null;
-        if (traceTokenManager != null) {
+        String token = request.getHeader(TRACETOKEN_HEADER);
```

Comment:
Add a comment that this is required because async responses fire on a different thread.